src/components/DateRangePicker.jsx

```diff
@@ -110,7 +111,12 @@ export default class DateRangePicker extends React.Component {
     } else if (focusedInput === END_DATE) {
       if (isInclusivelyBeforeDay(day, startDate)) {
```

Comment:
So this is not quite the logic we want in this if statement. I think the following is better:
```
const firstAllowedEndDate = startDate && startDate.clone().add(minimumNights, 'days');
if (!isInclusivelyAfterDay(day, firstAllowedEndDate)) {
  startDate = day;
  endDate = null;
} else {
  endDate = day;
  if (!startDate) {
    this.props.onFocusChange(START_DATE);
  } else {
    this.props.onFocusChange(null);
  }
}
```